editor/src/messages/tool/tool_messages/brush_tool.rs

```diff
@@ -255,78 +276,68 @@ impl Fsm for BrushToolFsmState {
 		tool_options: &Self::ToolOptions,
 		responses: &mut VecDeque<Message>,
 	) -> Self {
-		use BrushToolFsmState::*;
-		use BrushToolMessage::*;
-
-		let transform = document.document_legacy.root.transform;
+		let document_position = (document.document_legacy.root.transform).inverse().transform_point2(input.mouse.position);
+		let layer_position = tool_data.transform.inverse().transform_point2(document_position);
 
 		if let ToolMessage::Brush(event) = event {
 			match (self, event) {
-				(Ready, DragStart) => {
+				(BrushToolFsmState::Ready, BrushToolMessage::DragStart) => {
 					responses.add(DocumentMessage::StartTransaction);
-					let existing_points = load_existing_points(document);
-					let new_layer = existing_points.is_none();
-					if let Some((layer_path, points)) = existing_points {
-						tool_data.path = Some(layer_path);
-						//tool_data.set_image(image, responses);
-						if tool_data.points.is_empty() {
-							tool_data.points.push(points);
-						}
-					} else {
+					let layer_path = tool_data.load_existing_strokes(document);
+					let new_layer = layer_path.is_none();
+					if new_layer {
 						responses.add(DocumentMessage::DeselectAllLayers);
-						tool_data.path = Some(document.get_path_for_new_layer());
+						tool_data.layer_path = document.get_path_for_new_layer();
 					}
-
-					let pos = transform.inverse().transform_point2(input.mouse.position);
-
-					tool_data.points.push(vec![pos]);
+					let layer_position = tool_data.transform.inverse().transform_point2(document_position);
```

Comment:
@0HyperCube do you have any thoughts about this?

https://files.keavon.com/-/VibrantQuerulousAustralianfreshwatercrocodile/capture_8_.mp4